test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -0,0 +1,732 @@
+    GTEST_SKIP();
+  }
+
+  const std::string request_path_1("/name");
+
+  {
+    Http::TestResponseHeaderMapImpl response_headers{
+        {"date", formatter_.fromTime(time_source_.systemTime())},
+        {"cache-control", "public,max-age=3600"},
+        {":status", "200"},
+        {"etag", "\"foo\""},
+        {"content-length", "4"}};
+
+    ASSERT_THAT(insert(request_path_1, response_headers, "body"), IsOk());
+    auto lookup_context = lookup(request_path_1);
+    lookup_context->onDestroy();
+    ASSERT_NE(lookup_result_.headers_, nullptr);
+
+    // An age header is inserted by `makeLookupResult`
+    response_headers.setReferenceKey(Http::LowerCaseString("age"), "0");
+    EXPECT_THAT(lookup_result_.headers_.get(), HeaderMapEqualIgnoreOrder(&response_headers));
+  }
+
+  // Update the date field in the headers
+  time_source_.advanceTimeWait(Seconds(3601));
+
+  {
+    Http::TestResponseHeaderMapImpl response_headers =
+        Http::TestResponseHeaderMapImpl{{"date", formatter_.fromTime(time_source_.systemTime())},
+                                        {"cache-control", "public,max-age=3600"},
+                                        {":status", "200"},
+                                        {"etag", "\"foo\""},
+                                        {"content-length", "4"}};
+    updateHeaders(request_path_1, response_headers, {time_source_.systemTime()});
+    auto lookup_context = lookup(request_path_1);
+    lookup_context->onDestroy();
+
+    // An age header is inserted by `makeLookupResult`
+    response_headers.setReferenceKey(Http::LowerCaseString("age"), "0");
+    EXPECT_THAT(lookup_result_.headers_.get(), HeaderMapEqualIgnoreOrder(&response_headers));
+  }
+}
+
+TEST_P(HttpCacheImplementationTest, UpdateHeadersForMissingKey) {
+  const std::string request_path_1("/name");
+  Http::TestResponseHeaderMapImpl response_headers{
+      {":status", "200"},
+      {"date", formatter_.fromTime(time_source_.systemTime())},
+      {"cache-control", "public,max-age=3600"},
+      {"etag", "\"foo\""},
+  };
+  time_source_.advanceTimeWait(Seconds(3601));
+  updateHeaders(request_path_1, response_headers, {time_source_.systemTime()});
+  lookup(request_path_1);
+  EXPECT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+}
+
+TEST_P(HttpCacheImplementationTest, UpdateHeadersDisabledForVaryHeaders) {
+  if (!validationEnabled()) {
+    // UpdateHeaders would not be called when validation is disabled.
+    GTEST_SKIP();
+  }
+
+  const std::string request_path_1("/name");
+  const std::string time_value_1 = formatter_.fromTime(time_source_.systemTime());
+  Http::TestResponseHeaderMapImpl response_headers_1{{"date", time_value_1},
+                                                     {"cache-control", "public,max-age=3600"},
+                                                     {"accept", "image/*"},
+                                                     {"vary", "accept"}};
```

Comment:
:status is set in other response headers. Perhaps we should set it in all of the tests' response headers (there's a mix of whether we set it or not).

On that note, I wonder if we should extract a "makeTestResponseHeaders()" function or something to give a base set of headers that the individual tests modify.

Reply:
Put :status everywhere.

For now I think the base set of headers is too small for the helper to be worth it - it's just :status, and date for most but not all of the tests. So it'd save one or two lines for each test at the tradeoff of adding an extra step of indirection to read what the headers are, IMO not worth it.